### PR TITLE
libqxp: new recipe

### PR DESCRIPTION
**Pull request by nysnatuss**
Create recipe for [libqxp](https://wiki.documentfoundation.org/DLP/Libraries/libqxp), a library that parses the file format of QuarkXPress documents.

### CODE REVIEW

**Comment by nysnatuss** [comment]
@fbrosson Can you add the "wip-no-merge" tag? I currently don't have much free time than before to fix and experiment with the recipe. Also want to note that the recipe fails to build on my VM, so can you point out the mistakes I might made? :)

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
+
+libVersion=0.0.2
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-libVersion=0.0.2
+libVersion=0.0.1
```

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	libqxp$secondaryArchSuffix = $portVersion
+	lib:libqxp$secondaryArchSuffix = $libVersionCompat
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-	lib:libqxp$secondaryArchSuffix = $libVersionCompat
+	lib:libqxp_0.0$secondaryArchSuffix = $libVersionCompat
```
I found this by looking what haikuporter had left in `app-text/libqxp/work-0.0.1/packaging/libqxp/lib/`.

The `app-text/libqxp/work-0.0.1/packaging/` temporary directory is where haikuporter looks for what it has to compress into each hpkg, which are moved then to `app-text/libqxp/work-0.0.1/hpkgs/`.

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	haiku$secondaryArchSuffix
+	icu$secondaryArchSuffix
+	lib:libboost_system$secondaryArchSuffix
+	lib:libcppunit$secondaryArchSuffix+
+	lib:librevenge_0.0$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libqxp${secondaryArchSuffix}_devel = $portVersion
+	devel:libqxp$secondaryArchSuffix = $libVersion
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-	devel:libqxp$secondaryArchSuffix = $libVersion
+	devel:libqxp_0.0$secondaryArchSuffix = $libVersion
```

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm -f $libDir/*.la
+
+	prepareInstalledDevelLib libqxp
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-	prepareInstalledDevelLib libqxp
+	prepareInstalledDevelLib libqxp-0.0
```

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	cmd:awk
+	cmd:cmp
+	cmd:diff
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage libqxp$secondaryArchSuffix \
+	$libDir/libqxp.so.$libVersion
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
defineDebugInfoPackage libqxp$secondaryArchSuffix \
-	$libDir/libqxp.so.$libVersion
+	$libDir/libqxp-0.0.so.$libVersion
```

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	lib:libqxp$secondaryArchSuffix = $libVersionCompat
+	cmd:qxp2raw$commandSuffix = $portVersion
+	cmd:qxp2svg$commandSuffix = $portVersion
+	cmd:qxp2text$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	icu$secondaryArchSuffix
+	lib:libboost_system$secondaryArchSuffix
+	lib:libcppunit$secondaryArchSuffix+
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-	lib:libcppunit$secondaryArchSuffix+
+	lib:libcppunit$secondaryArchSuffix
```

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+documents. Currently it only understands documents created by QuarkXPress \
+3.1 to 4.1."
+HOMEPAGE="https://wiki.documentfoundation.org/DLP/Libraries/libqxp"
+COPYRIGHT="2017-2018 David Tardon, Aleksas Pantechovski"
+LICENSE="MPL v2.0"
+REVISION="1"
+SOURCE_URI="http://dev-www.libreoffice.org/src/libqxp/libqxp-$portVersion.tar.xz"
+CHECKSUM_SHA256="8c257f6184ff94aefa7c9fa1cfae82083d55a49247266905c71c53e013f95c73"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-ARCHITECTURES="x86_gcc2 x86 x86_64"
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
```
libqxp needs boost which is not available on x86_gcc2 primary arch.

File: app-text/libqxp/libqxp-0.0.1.recipe [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage libqxp$secondaryArchSuffix \
+	$libDir/libqxp.so.$libVersion
+
+BUILD()
+{
+	runConfigure ./configure --bindir=$commandBinDir --without-docs
```
Comment (fbrosson) [inline app-text/libqxp/libqxp-0.0.1.recipe]:
```diff
-	runConfigure ./configure --bindir=$commandBinDir --without-docs
+	runConfigure --omit-dirs binDir ./configure --bindir=$commandBinDir \
+		 --without-docs
```

**Pull request merged**